WordPress/Classes/Models/WPAccount+Lookup.swift

```diff
@@ -0,0 +1,157 @@
+    ///
+    @objc
+    var hasBlogs: Bool {
+        return !blogs.isEmpty
+    }
+
+    // MARK: - Object Lookups
+
+    /// Returns the default WordPress.com account, if one exists
+    ///
+    /// The default WordPress.com account is the one used for Reader and Notifications.
+    ///
+    static func lookupDefaultWordPressComAccount(in context: NSManagedObjectContext) throws -> WPAccount? {
+        guard let uuid = UserSettings.defaultDotComUUID, uuid.count > 0 else {
+            // No account, or no default account set. Clear the defaults key.
+            UserSettings.defaultDotComUUID = nil
+            return nil
+        }
+
+        return try lookup(withUUIDString: uuid, in: context)
+    }
+
+    /// Does a default WordPress.com account exist?
+    ///
+    /// The default WordPress.com account is the one used for Reader and Notifications.
+    ///
+    static func lookupHasDefaultWordPressComAccount(in context: NSManagedObjectContext) throws -> Bool {
+        return try lookupDefaultWordPressComAccount(in: context) != nil
+    }
+
+    /// Lookup a WPAccount by its local uuid
+    ///
+    /// - Parameters:
+    ///   - uuidString: The UUID (in string form) associated with the account
+    ///   - context: An NSManagedObjectContext containing the `WPAccount` object with the given `uuidString`.
+    /// - Returns: The `WPAccount` object associated with the given `uuidString`, if it exists.
+    ///
+    static func lookup(withUUIDString uuidString: String, in context: NSManagedObjectContext) throws -> WPAccount? {
+        let fetchRequest = NSFetchRequest<Self>(entityName: WPAccount.entityName())
+        fetchRequest.predicate = NSPredicate(format: "uuid = %@", uuidString)
+
+        guard let defaultAccount = try context.fetch(fetchRequest).first else {
+            return nil
+        }
+
+        /// This was brought over from the `AccountService`, but can (and probably should) be moved to an accessor for the property
+        if let displayName = defaultAccount.displayName {
+            defaultAccount.displayName = displayName.stringByDecodingXMLCharacters()
+        }
+
+        return defaultAccount
+    }
+
+    /// Lookup a WPAccount with the specified username, if it exists
+    ///
+    /// - Parameters:
+    ///   - username: The username associated with the account
+    ///   - context: An NSManagedObjectContext containing the `WPAccount` object with the given `username`.
+    /// - Returns: The `WPAccount` object associated with the given `username`, if it exists.
+    ///
+    static func lookup(withUsername username: String, in context: NSManagedObjectContext) throws -> WPAccount? {
+        let fetchRequest = NSFetchRequest<Self>(entityName: WPAccount.entityName())
+        fetchRequest.predicate = NSPredicate(format: "username = [c] %@ || email = [c] %@", username, username)
+        fetchRequest.includesPendingChanges = true // TODO: this shouldn't be necessary, it was copied from the `AccountService`
```

Comment:
Per the [docs](https://developer.apple.com/documentation/coredata/nsfetchrequest/1506724-includespendingchanges) the default is `true`.  Maybe just nuke it rather than add a comment to remind us to nuke it later? :)

Reply:
Nice ok! Yeah, I wasn't sure if there was any particular reason why its as there which was why I left the note, but definitely safe to remove here!

Removed in 561ccce352bdeeea3a6865b71955231f299f61ea